internal/services/aadgraph/application_app_role_resource.go

```diff
@@ -96,7 +97,7 @@ func applicationAppRoleResourceCreateUpdate(d *schema.ResourceData, meta interfa
 	} else {
 		rid, err := uuid.GenerateUUID()
 		if err != nil {
-			return fmt.Errorf("generating App Role for Object ID %q: %+v", objectId, err)
+			return tf.ErrorDiag(fmt.Sprintf("Generating App Role for application with object ID %q", objectId), err.Error(), "")
```

Comment:
I think we could make this a little cleaner, for ones that don't nee a "path"
```suggestion
			return tf.ErrorDiagF(err, "Generating App Role for application with object ID %q", objectId)
```